### PR TITLE
docs: improve multisig upgrade tutorial

### PR DESCRIPTION
**Pull request by JuArce**
# Improve multisig upgrade tutorial

## Description

This PR improves the deploy, upgrade and multisig approves tutorials

All the docs are in `docs/0_internal`

### CODE REVIEW

File: docs/0_internal/1_multisig_creation.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    ![Review](images/1_multisig_5.png)
+
+7. Confirm the transaction on your wallet.
+
+8. Wait until the transaction is confirmed. Then, you will see your Multisig wallet.
+
+    ![Multisig Wallet](images/1_multisig_6.png)
+
+After finishing this tutorial, you will have your Multisig wallet created. In this tutorial, we created a
```
Comment (uri-99) [inline docs/0_internal/1_multisig_creation.md]:
```suggestion
After finishing this tutorial, you will have your Multisig wallet created.
```

File: docs/0_internal/2_deploy_contracts.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
+# Deploying Aligned Contracts
+
+This guide will walk you through the deployment of the Aligned Layer contracts.
+
+Also, you will be able to deploy the Batch Payment Service contract.
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]:
```suggestion
Also, you will be able to deploy the Batcher Payment Service contract.
```

File: docs/0_internal/2_deploy_contracts.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+## Prerequisites
+
+- You need to have installed git and make.
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]:
```suggestion
- You need to have installed `git` and `make`.
```

File: docs/0_internal/2_deploy_contracts.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    ```shell
+    make install_foundry
+    foundryup -v nightly-a428ba6ad8856611339a6319290aade3347d25d9
+    ```
+
+## Eigenlayer Contracts
+
+To deploy Aligned contracts, you need previously deployed EigenLayer contracts.
+
+These contracts are not deployed by Aligned. Current EigenLayer contracts:
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]:
```suggestion
These contracts are not deployed by Aligned. These are the current EigenLayer contracts:
```

File: docs/0_internal/2_deploy_contracts.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+| `PRIVATE_KEY`                   | The private key of the account you want to deploy the contracts with. | <your_private_key>                                          | <your_private_key>                  |
+| `EXISTING_DEPLOYMENT_INFO_PATH` | The path to the file containing the deployment info about EigenLayer. | ./script/output/holesky/eigenlayer_deployment_output.json   | TBD                                 |
+| `DEPLOY_CONFIG_PATH`            | The path to the deployment config file for the Service Manager.       | ./script/deploy/config/holesky/aligned.holesky.config.json  | TBD                                 |
+| `OUTPUT_PATH`                   | The path to the file where the deployment info will be saved.         | ./script/output/holesky/alignedlayer_deployment_output.json | TBD                                 |
+| `ETHERSCAN_API_KEY`             | API KEY to verify the contracts in Etherscan.                         | <your_etherscan_api_key>                                    | <your_etherscan_api_key>            |
+
+You can find an example `.env` file in [.env.example.holesky](../../contracts/scripts/.env.example.holesky)
+
+> [!Warning]
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]:
```suggestion
> [!WARNING]
```

File: docs/0_internal/2_deploy_contracts.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+  "minimumStakes": [],
+  "strategyWeights": [],
+  "operatorSetParams": [],
+  "uri": ""
+}
+```
+
+You can find an example config file in [aligned.holesky.config.json](../../contracts/script/deploy/config/holesky/aligned.holesky.config.json).
+
+> [!Note]
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]:
```suggestion
> [!NOTE]
```

File: docs/0_internal/2_deploy_contracts.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+make deploy_aligned_contracts
+```
+
+Once the contracts are deployed, you will see the following output at `OUTPUT_PATH` file:
+
+```json
+{
+  "addresses": {
+    "alignedLayerProxyAdmin": "<aligned_layer_proxy_admin_address>",
+    "alignedLayerServiceManager": "<aligned_layer_service_manager_address>",
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]: missing pauser registry

File: docs/0_internal/2_deploy_contracts.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    "chainId": 17000,
+    "deploymentBlock": 1628199
+  },
+  "permissions": {
+    "alignedLayerAggregator": "<aligned_layer_aggregator_address>",
+    "alignedLayerChurner": "<aligned_layer_churner_address>",
+    "alignedLayerEjector": "<aligned_layer_ejector_address>",
+    "alignedLayerOwner": "<aligned_layer_owner_address>",
+    "alignedLayerUpgrader": "<aligned_layer_upgrader_address>",
+    "pauserRegistry": "<pauser_registry_address>"
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]: here it is AlignedLayerPauser, and the PauserRegistry goes on top bc it is a contract

File: docs/0_internal/2_deploy_contracts.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+  }
+}
+```
+
+## Batcher Payments Service Contracts
+
+This section will guide you through the deployment of the Aligned Layer contracts.
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]:
```suggestion
This section will guide you through the deployment of the Aligned Layer Batcher Payment Service contract.
```

File: docs/0_internal/2_deploy_contracts.md [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+Once you have configured the `.env` and `BATCHER_PAYMENT_SERVICE_CONFIG_PATH` files, you can run the following command:
+
+```bash
+make deploy_batcher_payment_service
+```
+
+Once the contracts are deployed, you will see the following output at `OUTPUT_PATH` file:
+
+```json
+{
```
Comment (uri-99) [inline docs/0_internal/2_deploy_contracts.md]: ditto

File: docs/0_internal/3_a_upgrade_contracts.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
+# Upgrade Contracts
+
+This guide is for upgrading contracts deployed without a multisig wallet. If you deployed the contract using a Multisig wallet, you can follow the [Upgrade Contracts with Multisig](./3_b_upgrade_contracts_with_multisig.md) tutorial.
```
Comment (uri-99) [inline docs/0_internal/3_a_upgrade_contracts.md]:
```suggestion
This guide is for upgrading contracts with a normal wallet.

If you deployed the contract using a Multisig wallet, you can follow the [Upgrade Contracts with Multisig](./3_b_upgrade_contracts_with_multisig.md) tutorial.
```

**Review by uri-99 — COMMENTED** [review]
WIP review

File: docs/0_internal/3_b_1_deploy_new_impl.md [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+> Make sure the `registryCoordinator` address is the same as the one you deployed in the [Deploy Contracts Guide](./2_deploy_contracts.md).
+>
+> Make sure the `registryCoordinatorImplementation` address is the same as the one you deployed in this guide.
+
+   
+## Next Steps
+
+Once you have deployed the new implementation of the contract you want to upgrade, you need to propose the upgrade transaction following this [guide](./3_b_2_propose_upgrade.md).
+
+Send the contracts addresses and `calldata` to the multisig participants to propose the upgrade transaction.
```
Comment (uri-99) [inline docs/0_internal/3_b_1_deploy_new_impl.md]:
```suggestion
Once you have deployed the new implementation of the contract you want to upgrade, you need to propose the upgrade transaction to the mulstisig, following this [guide](./3_b_2_propose_upgrade.md).

You must also send the contracts addresses and `calldata` you gathered from this guide, to all the multisig participants.
```

File: docs/0_internal/3_b_2_propose_upgrade.md [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+
+      If everything is correct, click on `Sign` to send the transaction.
+
+      ![Confirm transaction](./images/3_b_2_multisig_9.png)
+
+14. Now in your transactions, you should be able to see the newly created transaction.
+
+      ![New transaction](./images/3_b_2_multisig_10.png)
+
+15. If the transaction is correctly created, you have to wait until the required Multisig member signs the transaction to send it.
```
Comment (uri-99) [inline docs/0_internal/3_b_2_propose_upgrade.md]:
```suggestion
15. If the transaction is correctly created, you have to wait until the required Multisig member signs the transaction to send it.

A guide on how to sign the transaction can be found [here](./3_b_3_approve_upgrade.md)

```

**Review by uri-99 — APPROVED** [review]

File: Makefile [hunk 14]
```diff
@@ -563,10 +563,10 @@ run_storage: ## Run storage using storage-docker-compose.yaml
...
 	@docker compose -f storage-docker-compose.yaml up
 
-__DEPLOYMENT__:
-deploy_aligned_contracts: ## Deploy Aligned Contracts
-	@echo "Deploying Aligned Contracts..."
-	@. contracts/scripts/.env && . contracts/scripts/deploy_aligned_contracts.sh
+__DEPLOYMENT__: ## ____
+deploy_aligned_contracts: ## Deploy Aligned Contracts. Parameters: NETWORK=<mainnet|holesky|sepolia>
+	@echo "Deploying Aligned Contracts on $(NETWORK) network..."
+	@. contracts/scripts/.env.$(NETWORK) && . contracts/scripts/deploy_aligned_contracts.sh
```
Comment (MarcosNicolau) [inline Makefile:569]: Shouldn't we separate the `.env` based on the `NETWORK` on the `upgrade` targets as well?

File: docs/0_internal/3_b_1_deploy_new_impl.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+To deploy a new implementation, you can follow the steps below.
+
+## Prerequisites
+
+1. Make sure you have set variables as specified in the [Deploy Contracts Guide](./2_deploy_contracts.md).
+
+2. Add the following variables to the `.env` file:
+
+    ```makefile
+    MULTISIG=true
```
Comment (MarcosNicolau) [inline docs/0_internal/3_b_1_deploy_new_impl.md]: Following the comment on the`Makefile`, here we would use the `.env` used in the deployment. For example, if the deploy was in holesky, we'd need to pass `NETWORK=holesky` in the `make upgrade_contract.

**Review by MarcosNicolau — APPROVED** [review]
Documentation seemed really clear to me. Left a small comment on the `upgrade` guide, it is not necessary thou.

File: Makefile (hunk 14 above)
Comment (MauroToscano) [reply Makefile:569]: This is done

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**